### PR TITLE
Waze | waze.com

### LINKED ISSUE #3116: Waze | waze.com
**Prerequisites and essential questions** 
- [x] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** 
Maxou#1909

**What's the name of the service?** 
https://www.waze.com | Waze
(livemap)

**What should the Presence display?** 
Example for the livemap :
1. Waze
2. Planning a route
3. From [city] to [city]
4. Time elapsed

**If possible, please provide a logo for the service (512x512 minimum)** 

https://play-lh.googleusercontent.com/r7XL36PVNtnidqy6ikRiW1AHEIsjhePrZ8W5M4cNTQy5ViF3-lIDY47hpvxc84kJ7lw=s180-rw

### PR DESCRIPTION
**Pull request by enxg**
Closes #3116

Proof of work
image
image

### CODE REVIEW

File: websites/W/Waze/presence.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    presenceData.details = "Browsing";
+    presenceData.state = pages[path] ?? null;
+  }
+}
+fnd();
+setInterval(fnd, 5000);
+
+presence.on("UpdateData", async () => {
+
+
```
Comment (ririxidev) [inline websites/W/Waze/presence.ts:56]:
```suggestion
```

File: websites/W/Waze/presence.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+const presence = new Presence({
+    clientId: "854718080433520661"
+  }),
+  presenceData: PresenceData = {
+    largeImageKey: "logo",
+    startTimestamp: Date.now()
```
Comment (ririxidev) [inline websites/W/Waze/presence.ts:6]:
```suggestion
    startTimestamp: ~~(Date.now() / 1000)
```

File: websites/W/Waze/presence.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    start: string,
+    end: string;
+
+function fnd() {
+  path = window.location.pathname;
+  if (!path.endsWith("/")) path = `${path}/`;
+  path = path.replace(/(\/[a-z]{2}|\/[a-z]{2}-[A-Z]{2})\//,"/");
+
+  if (path.includes("/live-map")) {
+    start = end = "";
```
Comment (ririxidev) [inline websites/W/Waze/presence.ts:39]: eee what? you don't need this line?
```suggestion
```

**Review by rxri — CHANGES_REQUESTED** [review]
i hope that sites after logging in are supported too.

File: websites/W/Waze/presence.ts (hunk 3 above)
Comment (RedS-DEV) [reply websites/W/Waze/presence.ts:39]: I needed it before, but I changed the code. Maybe I don't need it now.

**Comment by RedS-DEV** [comment]
> i hope that sites after logging in are supported too.

You said only livemap on issue, I made the livemap and added some pages.

File: websites/W/Waze/presence.ts (hunk 3 above)
Comment (ririxidev) [reply websites/W/Waze/presence.ts:39]: I don't think you need that since you have `let start: string, end: string` already 😂

**Comment by ririxidev** [comment]
Well okay whatever

File: websites/W/Waze/presence.ts (hunk 3 above)
Comment (RedS-DEV) [reply websites/W/Waze/presence.ts:39]: No, it was for clearing the origin and destination.

Comment (ririxidev) [reply websites/W/Waze/presence.ts:39]: Okay then

Comment (RedS-DEV) [reply websites/W/Waze/presence.ts:39]: Tested it now and it is not needed anymore because of how I changed the code.

**Review by rxri — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**